lib/util.js

```diff
@@ -1,15 +1,22 @@
 const util = require('util');
-const fs = require('fs');
+const path = require('path');
+const which = require('which');
+const { pathExists, readFile } = require('fs-extra');
 
 const childProcess = require('child_process');
 
 const { promisify } = util;
 const exec = promisify(childProcess.exec);
-const readFile = promisify(fs.readFile);
 const wait = promisify(setTimeout);
 
+async function packageExist(pckg, dir) {
+  return pathExists(path.resolve(dir, `node_modules/.bin/${pckg}`));
```

Comment:
This fails if the package is installed in the dependencies of a parent directory, which is still considered as available by both node and npx (that's why we can run `npx lerna` from within any of the subpackages in our monorepo).

There are two options around this:
- Using `require.resolve('zos/index')` to look for the package root, and then walk to the binary (note that `.bin` folders are not understood by `require`, so you need to look for a source file and then move onto the corresponding bin)
- Using `findUp` to look for the binary `node_modules/.bin/${pckg}` in any parent folder (see [here](https://github.com/zeppelinos/zos/pull/914/commits/0eb7c27132a4aae1ca7eeb217ea9289f0495087a#diff-f948b1ff00474c4075ca089abb09ac49R38) for an example)

Reply:
Alternatively, it seems like [this package](https://www.npmjs.com/package/npm-which) takes care of it.